registry/proxy/proxyauth.go

```diff
@@ -33,16 +33,42 @@ func (c credentials) SetRefreshToken(u *url.URL, service, token string) {
 }
 
 // configureAuth stores credentials for challenge responses
-func configureAuth(username, password string) (auth.CredentialStore, error) {
-	creds := map[string]userpass{
-		tokenURL: {
+func configureAuth(username, password, remoteURL string) (auth.CredentialStore, error) {
+	creds := map[string]userpass{}
+	authURLs, err := getAuthURLs(remoteURL)
+
```

Comment:
nit: can you put this empty line above the get call so the err check is on the next line